src/util/worker/mod.rs

```diff
@@ -111,7 +124,11 @@ impl<T: Display> Scheduler<T> {
     /// If the worker is stopped, an error will return.
     pub fn schedule(&self, task: T) -> Result<(), Stopped<T>> {
         debug!("scheduling task {}", task);
-        if let Err(SendError(Some(t))) = self.sender.send(Some(task)) {
+        let send_result = match self.sender {
+            TaskSender::Unbounded(ref tx) => tx.send(Some(task)),
+            TaskSender::Bounded(ref tx) => tx.send(Some(task)),
```

Comment:
`schedule` should never block.

Reply:
So we don't need `try_schedule`, and `schedule` should return `Result<(), TrySendError>`?

Reply:
I think we can let `schedule` block if we have set `pending_capacity`. If we don't want any block, we can call `try_schedule`.

Reply:
There is no such use case in our system that would like to block and wait for the messages being sent. Block is implicit and unpredictable, should be avoided generally.